src/benchmark_register.cc

```diff
@@ -461,16 +463,78 @@ int Benchmark::ArgsCnt() const {
   return static_cast<int>(args_.front().size());
 }
 
+void Benchmark::Init() {}
+
+namespace {
+
 //=============================================================================//
 //                            FunctionBenchmark
 //=============================================================================//
 
-void FunctionBenchmark::Run(State& st) { func_(st); }
+// The class used to hold all Benchmarks created from static function.
+// (ie those created using the BENCHMARK(...) macros.
+class FunctionBenchmark : public Benchmark {
+ public:
+  FunctionBenchmark(const char* name, Function* func)
+      : Benchmark(name), func_(func) {}
+
+  void Run(State& st) override { func_(st); }
+
+ private:
+  Function* func_;
+};
+
+//=============================================================================//
```

Comment:
just to mention it, all those comments are 81 characters width(not including the newline)